android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -186,6 +190,32 @@ constructor(
               }
             }
         }
+
+        // Set configured openSrpId on Questionnaire
+        questionnaireConfig.uniqueIdAssignment?.let { uniqueIdAssignmentConfig ->
+          find(uniqueIdAssignmentConfig.linkId)?.apply {
+            // Extract ID from a Group, should be modified in future to support other resources
+            val resource =
+              defaultRepository.retrieveUniqueIdAssignmentResource(
+                questionnaireConfig.uniqueIdAssignment,
+              )
+
+            val extractedId =
+              fhirPathDataExtractor.extractValue(
+                base = resource,
+                expression = uniqueIdAssignmentConfig.idFhirPathExpression,
+              )
+            if (resource != null && extractedId.isNotEmpty()) {
+              uniqueIdResourcePair = Pair(extractedId, resource) // To be updated upon submission
```

Comment:
since we're looking at the QR, do we need to set the state here? I'd simply not set the state if we do not have to, if we do need this in `retireUsedQuestionnaireUniqueId`, can we pass it as an argument?

Reply:
Yes, this state is not necessary. The ID can be obtained from the QuestionnaireResponse.